### ISSUE TITLE
Bookmarks

### ISSUE THREAD

**Issue by Lucifie** [gagax1234/OpenFunscripter#12]
I would like to create a bookmark tag in the funscript format that everyone can use. Maybe we can do something similar like the metadata tag :)

**Comment by gagax1234** [comment]
Well then if you want to specify one go right ahead.

**Comment by Lucifie** [comment]
I will have a look and report back later :)

**Comment by Lucifie** [comment]
I have made a first proposal. See the end of the funscript file for an example. It's a nested structure where you can use the chapter object to create chapters and sub chapters. The root chapter is the whole script with its first and last timestamp. Then the chapters array contains the first level.

I used the tag _chapters_ since _bookmark_ was already taken by the original funscript format.

Rename the file extension to funscript: [1.txt](https://github.com/gagax1234/OpenFunscripter/files/6260189/1.txt)

### chapter
- **name** string=""
- **start** long=0
- **end** long=0
- **chapters** = chapter()

![grafik](https://user-images.githubusercontent.com/81717113/113615704-7a683100-9654-11eb-9484-862adb7c92ca.png)

**Comment by gagax1234** [comment]
**TL;DR**: I don't want to export anything "unofficial" anymore to the ".funscript" files which I'm saving.
The only thing which I'm willing to support is the metadata but even that is no where **publicly** visible specified.
The rest is really just my trail of thought and how I come to that conclusion.

Uhm yeah. I'm definetely not extending the ability to create bookmarks in OFS beyond what I'm already doing.
In general don't see what this has to do with funscript creation at all.
Bookmarks are supposed to aid the scripting process.

Why create another tag in the root and not under "metadata"?
The "original funscript format" (if there is such a thing) never had a "bookmark" tag that's a JFS invention.
The original format just looks like this. Source: [pkg.go.dev/github.com/funjack/launchcontrol/protocol/funscript](https://pkg.go.dev/github.com/funjack/launchcontrol/protocol/funscript?utm_source=godocalready)
```json
{
	"version": "1.0",
	"inverted": false,
	"range": 90,
	"actions": [
		{"pos": 0, "at": 100},
		{"pos": 100, "at": 500},
	]
}
```
If someone was to maintain an actual specification you'd know.
But there's is no specification (except for this [pkg.go.dev/github.com/funjack/launchcontrol/protocol/funscript](https://pkg.go.dev/github.com/funjack/launchcontrol/protocol/funscript?utm_source=godocalready)) which makes me regret agreeing to this initially because I don't want to do it anymore.
I'm making a push to move away from ".funscript" and will create my own project format for OFS because while it was convenient for me to store my internal settings there I think it did more damage than good because of all the scripts which now float around with that junk in it.
I'm sorry but that's how I feel about.

**Comment by Lucifie** [comment]
I think it's ok to include the _chapters_ key in the metadata so we don't have to create a new one on the top level.

I can understand the need for a separate project file and thats ok. But if you discard the metadata on export then you will cut of data which is already available and which the user can use. As a script creator I would like to keep this data in the file so other people can see it. Using a second tool to add the data again seems wrong.

I played around with the test file and made a little preview in HandyControl. For navigation this is realy nice and shows what is happening in the script.

_Script with chapters_
![grafik](https://user-images.githubusercontent.com/81717113/113779561-347f9b80-972e-11eb-97ce-01f2bd1cec0f.png)
_Script without chapters_
![grafik](https://user-images.githubusercontent.com/81717113/113779811-8de7ca80-972e-11eb-88a4-9b10550c43c2.png)

If you are not going to help then nobody can. There is no other scripting app available to the public that can include it. So in the end I need to create my own tag in the script to make this happen.

Script players can just ignore tags in the scripts so even if your OFS data is included this no problem at all. There are also tags from JFS which are included and they don't make any problems.
